packages/docusaurus-theme-classic/src/theme/NavbarItem/DefaultNavbarItem.tsx

```diff
@@ -159,7 +159,7 @@ function NavItemMobile({
   const menuListRef = useRef<HTMLUListElement>(null);
   const {pathname} = useLocation();
   const [collapsed, setCollapsed] = useState(
-    () => !items?.some((item) => isSamePath(item.to, pathname)) ?? true,
+    () => !items?.filter((item) => item.to).some((item) => isSamePath(item.to, pathname)) ?? true,
```

Comment:
This now filters out all the navbar links with `href` instead of `to` so that we don't get undefined down the line. Path normalization is not required on external links I assume? Or instead of passing further `item.to`, you have to path just `item` and later check the path for `to` and `href`